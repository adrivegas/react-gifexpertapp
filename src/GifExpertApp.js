import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Gamora', 'Iron Man', 'Thor']);

    const handleAdd = () => {
        // setCategories([...categories, 'Visión']);
        // También lo puedo hacer con un callback, devolviendo un nuevo arreglo con todas las categorías
        setCategories(cats => [...cats, 'Visión']); 
    };

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr/>

            <ol>
                {
                    categories.map( category => {
                        return <li key= { category }> {category} </li>
                    })
                }
            </ol>
        </div>
    )
}
