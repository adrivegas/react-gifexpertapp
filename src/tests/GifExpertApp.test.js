import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {

    test('se debe mostrar el componente correctamente ', () => {

        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('debe de mostrar una lista de categorÃ­as', () => {
        
        const categories = ['Gamora', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    })
    
    
    
})
