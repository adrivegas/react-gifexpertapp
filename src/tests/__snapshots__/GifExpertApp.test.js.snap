// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Gamora"
      key="Gamora"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> se debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
